integration-tests/mock.ts

```diff
@@ -0,0 +1,51 @@
+import assert from "assert";
+import { blue, red } from "./base";
+import wcmatch from "wildcard-match";
+
+interface MockedFetch {
+    method: "GET" | "POST" | "PUT" | "PATCH";
+    url: string; // use some/path/* as a wildcard
+    body?: string; // Use * as a wildcard
+    responseStatus: number;
+    response?: string | object;
+}
+
+let expectedCalls: MockedFetch[] = [];
+
+export function expectFetch(mocked: MockedFetch) {
+    expectedCalls.push(mocked);
+}
+
+export function setupFetchMock() {
+    const originalFetch = global.fetch;
+
+    async function mockedFetch(resource: RequestInfo, options: RequestInit) {
+        const url = "" + resource;
+
+        const mock = expectedCalls.find(it => (it.url === url || wcmatch(it.url)(url)) && it.method === (options.method ?? "GET"));
```

Comment:
What if there are multiple matching results within the list? It could be on purpose, but especially with wildcards it's hard to predict what's used in which call.

Reply:
Well I wasn't sure if the order is always deterministic, but I guess we can take care of that once we have that case. Changed this to make it strictly sequential (and easier to follow).